localstack/services/sqs/query_api.py

```diff
@@ -0,0 +1,142 @@
+
+
+@route(
+    '/<regex("[0-9]{12}"):account_id>/<regex("[a-zA-Z0-9_-]+(.fifo)?"):queue_name>',
+    methods=["POST", "GET"],
+)
+def legacy_handler(request: Request, account_id, queue_name) -> Response:
+    # previously, Queue URLs were created as http://localhost:4566/000000000000/my-queue-name. Because the region is
+    # ambiguous in this request, we fall back to the default region and hope for the best.
+    return handle_request(request, config.DEFAULT_REGION)
+
+
+class UnknownOperationException(Exception):
+    pass
+
+
+class InvalidAction(CommonServiceException):
+    def __init__(self, action: str):
+        super().__init__(
+            "InvalidAction",
+            f"The action {action} is not valid for this endpoint.",
+            400,
+            sender_fault=True,
+        )
+
+
+class BotoException(CommonServiceException):
+    def __init__(self, boto_response):
+        error = boto_response["Error"]
+        super().__init__(
+            code=error.get("Code", "UnknownError"),
+            status_code=boto_response["ResponseMetadata"]["HTTPStatusCode"],
+            message=error.get("Message", ""),
+            sender_fault=error.get("Type", "Sender") == "Sender",
+        )
+
+
+def handle_request(request: Request, region: str) -> Response:
+    if request.is_json:
+        # TODO: the response should be sent as JSON response
+        raise NotImplementedError
+
+    try:
+        response, operation = try_call_sqs(request, region)
+        return serializer.serialize_to_response(response, operation)
+    except UnknownOperationException:
+        return Response("<UnknownOperationException/>", 404)
+    except CommonServiceException as e:
+        # use a dummy operation for the serialization to work
+        op = service.operation_model(service.operation_names[0])
+        return serializer.serialize_error_to_response(e, op)
+    except Exception as e:
+        LOG.exception("exception")
+        op = service.operation_model(service.operation_names[0])
+        return serializer.serialize_error_to_response(
+            CommonServiceException(
+                "InternalError", f"An internal error ocurred: {e}", status_code=500
+            ),
+            op,
+        )
+
+
+def try_call_sqs(request: Request, region: str) -> Tuple[Dict, OperationModel]:
+    action = request.values.get("Action")
+    if not action:
+        raise UnknownOperationException()
+
+    if action in ["ListQueues", "CreateQueue"]:
+        raise InvalidAction(action)
+
+    # prepare aws request
+    params = {"QueueUrl": request.base_url}
+    # if a QueueUrl is already set in the body, it should overwrite the one in the URL
+    params.update(request.values)
+    body = urlencode(params)
+
+    try:
+        operation, service_request = parser.parse(
+            Request("POST", "/", headers=request.headers, body=body)
+        )
+    except OperationNotFoundParserError:
+        raise InvalidAction(action)
```

Comment:
I find it interesting that the parser always takes a POST request, regardless of what the original call was. I am assuming that is intentional, just making sure :)

Reply:
yes actually, great catch! sqs uses the AWS query protocol, and in the query protocol each operation is a POST request whether the request is encoded as a query string in the request body. it's just how the protocol is specified. with the query API, we can make GET requests, so we translate them to POST requests to the backend API.